<launch>
<!-- A DJI Bag (with GPS data) is published. Color images taken by nap. nap
publishesh co-location messages. GPS2odom node publishes odometry messages
from gps.latitude and gps.longitude. colocation_viz subscribes to
colocation messages and odometry to produce rviz results.
-->

<!-- Input Topics -->
<arg name="color_image_topic" default="/dji_sdk/image_raw"/>

<arg name="bag_odometry_topic" default="/gps_odom"/>


<!-- TODO : set these mapto -->
<arg name="raw_node_topic" default="/raw_graph_node" />
<arg name="raw_edge_topic" default="/raw_graph_edge" />


<!-- UST Aerial Seq -->
<arg name="bag_base" default="/media/mpkuse/Bulk_Data/ros_bags/rec/"/>
<arg name="bag_name" default="22.bag"/>
<node name="rosbag" pkg="rosbag" type="play" args="-d 10 $(arg bag_base)/$(arg bag_name)" output="log" />




<!-- NAP Node -->

<!--<node name="nap" pkg="nap" type="nap_time_node.py" output="screen" >-->

<!--<node name="nap" pkg="nap" type="nap_geom_node.py" output="log" >-->
<node name="nap" pkg="nap" type="nap_graph_modelling.py" output="screen" >
  <remap to="$(arg color_image_topic)" from="/semi_keyframes"/>
</node>


<!-- DBOW Node -->
<!--
<node name="dbow_node" pkg="nap" type="dbow3_naive" output="screen">
  <remap to="$(arg color_image_topic)" from="/color_image_inp" />
</node>
-->

<!-- GPS Odometry Node -->
<node name="gps2odometry" pkg="nap" type="gps2odometry.py" output="log">
    <remap to="/dji_sdk/global_position" from="/dji_sdk/global_position" />
</node>

<!-- Write Odometry to file(debug) -->
<node name="odometry2file" pkg="nap" type="odometry2file.py" output="log">
  <remap from="/vins_estimator/odometry" to="$(arg bag_odometry_topic)" />
</node>


<!-- Colocation visualizer -->
<node name="colocation_viz2" pkg="nap" type="colocation_viz2.py" output="log">
    <remap from="/vins_estimator/odometry" to="$(arg bag_odometry_topic)" />
</node>


<node name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find nap)/rviz/viz_gps.rviz" />



</launch>
